lib/src/testing_library/dom/queries/by_testid.dart

```diff
@@ -0,0 +1,136 @@
+/// https://testing-library.com/docs/queries/bytestid/
+@JS()
+library over_react_test.src.testing_library.dom.queries.by_testid;
+
+import 'dart:html' show Element, promiseToFuture;
+
+import 'package:js/js.dart';
+
+import 'package:over_react_test/src/testing_library/dom/async/types.dart';
+import 'package:over_react_test/src/testing_library/dom/async/wait_for.dart';
+import 'package:over_react_test/src/testing_library/dom/matches/types.dart';
+import 'package:over_react_test/src/testing_library/dom/queries/interface.dart';
+
+/// PRIVATE. Do not export from this library.
+///
+/// The public API is either the top level function by the same name as the methods in here,
+/// or the methods by the same name exposed by `screen` / `within()`.
+mixin ByTestIdQueries on IQueries {
```

Comment:
One thing to note here is that the JS library defaults to using `data-testid`, but all users of the existing over_react_test stuff are using `data-test-id`. We would likely want to figure out a way to make our current default - be the default in some sort of global setup / call to `config()`.